src/main/java/io/ebeaninternal/server/query/CQueryBuilder.java

```diff
@@ -377,11 +377,11 @@ private SqlTree createNativeSqlTree(OrmQueryRequest<?> request, CQueryPredicates
       PreparedStatement statement = connection.prepareStatement(sql);
       predicates.bind(statement, connection);
 
-      List<String> propertyNames = new ArrayList<>();
-
       ResultSet resultSet = statement.executeQuery();
       ResultSetMetaData metaData = resultSet.getMetaData();
+
       int cols = 1 + metaData.getColumnCount();
+      List<String> propertyNames = new ArrayList<>(cols - 1);
       for (int i = 1; i < cols; i++) {
```

Comment:
@rbygrave I also noticed this method when debugging and was wondering why you add +1 to cols. I would suggest to change the comparison in the for loop from "<" to "<=" and remove all the +1's and -1's.